Link edition is broken on Pangea 3 since merge of PR #3219

**Describe the bug**
The compilation of some of the unit tests fails at link edition step due to ` multiple definition of` functions.

**To Reproduce**
Steps to reproduce the behavior:
1. configure GEOS with commit `18db014` and master branch of the TPLs (commit `0fd15c395`)
2. build, for example, the `testAquiferBoundaryCondition` test case
3. See error

```
[100%] Linking CXX executable ../../../tests/testAquiferBoundaryCondition
../../../lib/libcommon.a(BufferAllocator.cpp.o): In function `(anonymous namespace)::anonymous_chai::pluginLinker::pluginLinker()':
/appli_RD/FROEHLY/thirdPartyLibs/install-pangea3-gcc8.4.1-openmpi-4.1.2-debug/chai/include/umpire/util/backtrace.inl:38: multiple definition of `umpire::util::(anonymous namespace)::backtrace_enabled()'
../../../lib/libcommon.a(BufferAllocator.cpp.o):/appli_RD/FROEHLY/thirdPartyLibs/install-pangea3-gcc8.4.1-openmpi-4.1.2-debug/chai/include/umpire/util/backtrace.inl:38: first defined here
../../../lib/libcommon.a(BufferAllocator.cpp.o): In function `umpire::util::(anonymous namespace)::build_backtrace()':
/appli_RD/FROEHLY/thirdPartyLibs/install-pangea3-gcc8.4.1-openmpi-4.1.2-debug/chai/include/umpire/util/backtrace.inl:60: multiple definition of `umpire::util::(anonymous namespace)::build_backtrace()'
../../../lib/libcommon.a(BufferAllocator.cpp.o):/appli_RD/FROEHLY/thirdPartyLibs/install-pangea3-gcc8.4.1-openmpi-4.1.2-debug/chai/include/umpire/util/backtrace.inl:60: first defined here
../../../lib/libcommon.a(BufferAllocator.cpp.o): In function `umpire::util::(anonymous namespace)::stringify[abi:cxx11](std::vector<void*, std::allocator<void*> > const&)':
...
```

**Expected behavior**
The compilation succeed / no error at link edition

**Platform (please complete the following information):**
- Machine: Pangea 3
- Architecture: `ppc64le`
- Compiler: `gcc-8.4.1` + `nvcc-11.5.50`
- Host-config file: `TOTAL/pangea3-gcc8.4.1-openmpi-4.1.2.cmake`
- GEOSX Version: since commit 18db014

Also linked to the PR #3159 that is broken since the last modifications in the develop branch.

--- a/src/coreComponents/unitTests/constitutiveTests/CMakeLists.txt
+++ b/src/coreComponents/unitTests/constitutiveTests/CMakeLists.txt
@@ -38,7 +38,6 @@ if ( GEOS_BUILD_SHARED_LIBS )
   list( APPEND dependencyList geosx_core HDF5::HDF5 )
 else()
   list( APPEND dependencyList ${geosx_core_libs} HDF5::HDF5 )
-  list( APPEND dependencyList ${geosx_core_libs} HDF5::HDF5 )
 endif()
 
 if( ENABLE_PVTPackage )

--- a/src/coreComponents/unitTests/fieldSpecificationTests/CMakeLists.txt
+++ b/src/coreComponents/unitTests/fieldSpecificationTests/CMakeLists.txt
@@ -11,7 +11,6 @@ if ( GEOS_BUILD_SHARED_LIBS )
   list( APPEND dependencyList geosx_core HDF5::HDF5 )
 else()
   list( APPEND dependencyList ${geosx_core_libs} HDF5::HDF5 )
-  list( APPEND dependencyList ${geosx_core_libs} HDF5::HDF5 )
 endif()
 
 if (TARGET pugixml::pugixml)

--- a/src/coreComponents/unitTests/finiteVolumeTests/CMakeLists.txt
+++ b/src/coreComponents/unitTests/finiteVolumeTests/CMakeLists.txt
@@ -8,7 +8,6 @@ if ( GEOS_BUILD_SHARED_LIBS )
   list( APPEND dependencyList geosx_core HDF5::HDF5 )
 else()
   list( APPEND dependencyList ${geosx_core_libs} HDF5::HDF5 )
-  list( APPEND dependencyList ${geosx_core_libs} HDF5::HDF5 )
 endif()
 
 # Add gtest C++ based tests

--- a/src/coreComponents/unitTests/fluidFlowTests/CMakeLists.txt
+++ b/src/coreComponents/unitTests/fluidFlowTests/CMakeLists.txt
@@ -11,7 +11,6 @@ if ( GEOS_BUILD_SHARED_LIBS )
   list( APPEND dependencyList geosx_core HDF5::HDF5 )
 else()
   list( APPEND dependencyList ${geosx_core_libs} HDF5::HDF5 )
-  list( APPEND dependencyList ${geosx_core_libs} HDF5::HDF5 )
 endif()
 
 set( dependencyList testingUtilities )

--- a/src/externalComponents/CMakeLists.txt
+++ b/src/externalComponents/CMakeLists.txt
@@ -25,7 +25,7 @@ if( ${BUILD_SHARED_LIBS} )
   set( externalComponentsLinkList ${externalComponentsList} )
 else()
  foreach(component ${externalComponentsList})
-   set( externalComponentsLinkList ${externalComponentsLinkList} ${GEOSX_LINK_PREPEND_FLAG} ${component} ${GEOSX_LINK_POSTPEND_FLAG} )
+   set( externalComponentsLinkList ${externalComponentsLinkList} ${GEOS_LINK_PREPEND_FLAG} ${component} ${GEOS_LINK_POSTPEND_FLAG} )
  endforeach()
 endif()
 

